std/protocols/permutation.asm

```diff
@@ -84,29 +81,17 @@ let permutation: expr, expr[], Fp2<expr>, Fp2<expr>, Constr -> Constr[] = |is_fi
 
     let (lhs_selector, lhs, rhs_selector, rhs) = unpack_permutation_constraint(permutation_constraint);
 
-    let _ = assert(len(lhs) == len(rhs), || "LHS and RHS should have equal length");
-    let _ = if !is_extension(acc) {
-        assert(!needs_extension(), || "The Goldilocks field is too small and needs to move to the extension field. Pass two accumulators instead!")
-    } else { };
-
     // On the extension field, we'll need two field elements to represent the challenge.
     // If we don't need an extension field, we can simply set the second component to 0,
     // in which case the operations below effectively only operate on the first component.
```

Comment:
Same as in the lookup.asm, we can move this explanation to `fp2_from_array`, and delete this